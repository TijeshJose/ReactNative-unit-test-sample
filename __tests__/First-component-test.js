import 'react-native';
import React from 'react';
import FirstComponent from '../src/component/first';

import renderer from 'react-test-renderer';

//check first page renders correctly
it('First page component renders correctly', () => {
    const tree = renderer.create(<FirstComponent name={'Test'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

//check first component receives props correctly
it('Check props received correctly', () => {
    const instance = renderer.create(<FirstComponent name={'Test'} />).root;
    expect(instance.props.name).toEqual('Test');
});