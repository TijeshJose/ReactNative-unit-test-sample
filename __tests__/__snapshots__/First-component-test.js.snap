// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First page component renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginTop": 20,
      }
    }
  >
    Test
  </Text>
</View>
`;
